libcloud/compute/drivers/openstack.py

```diff
@@ -4441,6 +4510,64 @@ def __repr__(self):
         )
 
 
+class OpenStack_2_FloatingIpAddress(OpenStack_1_1_FloatingIpAddress):
+    """
+    Floating IP info 2.0.
+    """
+
+    def __init__(self, id, ip_address, pool, node_id=None, driver=None, extra=None):
+        self.id = str(id)
+        self.ip_address = ip_address
+        self.pool = pool
+        self.node_id = node_id
+        self.driver = driver
+        self.extra = extra if extra else {}
+
+    def get_pool(self):
+        if not self.pool:
+            try:
+                # If pool is not set, get the info from the floating_network_id
+                net = self.driver.ex_get_network(self.extra["floating_network_id"])
+            except Exception:
+                net = None
+            if net:
+                self.pool = OpenStack_2_FloatingIpPool(
+                    net.id, net.name, self.driver.network_connection
+                )
+        return self.pool
+
+    def get_node_id(self):
+        if not self.node_id:
+            # if not is not set, get it from port_details
```

Comment:
"if not" > "if node_id"

Reply:
I don't get your point here ...

Reply:
Sorry. This phrase "if not is not set" - I think you meant "if node_id is not set".